framework_tvb/tvb/tests/framework/adapters/simulator/simulator_adapter_test.py

```diff
@@ -112,38 +116,34 @@ def test_happy_flow_launch(self, connectivity_factory, operation_factory):
 
         SIMULATOR_PARAMETERS['connectivity'] = self.connectivity.gid
 
-        OperationService().initiate_prelaunch(self.operation, self.simulator_adapter, **SIMULATOR_PARAMETERS)
+        OperationService().initiate_prelaunch(self.operation, self.simulator_adapter, self.simulator_with_connectivity_model())
         sim_result = dao.get_generic_entity(TimeSeriesRegion, 'TimeSeriesRegion', 'type')[0]
         assert sim_result.read_data_shape() == (32, 1, self.CONNECTIVITY_NODES, 1)
 
-    def _estimate_hdd(self, new_parameters_dict):
-        """ Private method, to return HDD estimation for a given set of input parameters"""
-        filtered_params = self.simulator_adapter.prepare_ui_inputs(new_parameters_dict)
-        self.simulator_adapter.configure(**filtered_params)
-        return self.simulator_adapter.get_required_disk_size(**filtered_params)
+    def simulator_with_connectivity_model(self):
+        simulator_adapter_model = self.simulator_adapter.get_view_model()()
```

Comment:
Why not import `SimulatorAdapterModel` directly, and create an instance of that class?
We do not need to be polymorphic here

Reply:
Fixed it as by your suggestion :)